desired.go

```diff
@@ -0,0 +1,9 @@
+package service
+
+import (
+	"context"
+)
+
+func (r *Resource) GetDesiredState(ctx context.Context, obj interface{}) (interface{}, error) {
+	return r.desiredServicesFunc(ctx, obj)
```

Comment:
Please lets keep the calls explicit for our error masking. The masking tells a story via its stack. This story has to match the real call graph and your mental model. This here breaks it.